engine/ekf.py

```diff
@@ -101,14 +103,61 @@ def get_h_jac(self, pose):
         """
         return np.array([[1, 0, 0], [0, 1, 0], [0, 0, 1]])
 
-    def predict_step(self, control):
+    def get_arc_lengths(self, left_side_w, right_side_w):
```

Comment:
is this under the assumption that the time step is 1?

Reply:
im not really sure how to incorporate the timestep - ive seen it used in other areas of the code - im not sure if its the same throughout or if each one has their own time-cycle --- open to suggestions here

Reply:
I think timestep is the cycle time, so effectiively how long we wait before changing the input voltage to the motors (in the while loop there's a delay of 20ms and changing input voltage to motor changes once per while loop iteration? so I think that'll mean the time step is 20ms). Even if tilmestep changes, it might be a good option to have it as a param and make the user define it

Reply:
resolving this conversation - timestep will be determined by the caller